lib/cloudapp/cli/collections.rb

```diff
@@ -0,0 +1,104 @@
+require "http"
+require "pry"
+require "terminal-table"
+
+module CloudApp
+  module CLI
+    class Collections
+      COLLECTION_FIELDS = ["id", "name", "shared", "is_public", "created_at", "updated_at"]
+      ITEM_FIELDS = ["id", "slug", "name", "item_type", "view_counter", "favorite", "share_url", "created_at", "updated_at"]
+
+      def initialize(token)
+        @BASE_URL = "https://my.cl.ly/v4/collections"
+        @HTTP = HTTP.auth("Bearer #{token}")
+          .with(accept: "application/json", content_type: "application/json")
+          .accept(:json)
+      end
+
+      def self.token_to_jwt(token)
+        http = HTTP.with(accept: "application/json", content_type: "application/json")
+          .accept(:json)
+        JSON.parse(http.post("https://my.cl.ly/v3/jwt_tokens?token=#{token}").body.readpartial)["jwt"]
+      end
+
+      def list_collections
+        resp = JSON.parse(@HTTP.get(@BASE_URL).body.readpartial)
+        collections = []
+
+        resp.each do |h|
+          h.select! { |k, v| COLLECTION_FIELDS.include? k }
+          collections << h.values
+        end
+
+        table = Terminal::Table.new :title => "Listing Collections", :headings => ["Collection ID", "Collection Name", "Shared", "Public", "Created At", "Updated At"], :rows => collections
+        puts table
+      end
+
+      def list_items(collection_id)
+        resp = JSON.parse(@HTTP.get("#{@BASE_URL}/#{collection_id}").body.readpartial)
+        items = []
+        resp["items"].each do |i|
```

Comment:
What do you think about:

```ruby
items = resp["items"].map{ |i| i.slice(*ITEM_FIELDS).values }

Reply:
Rails is required for Hash#slice

Reply:
Ahh, you're right... active support hurts our brains.. 😧